vayesta/ewf/urdm.py

```diff
@@ -208,11 +211,494 @@ def make_rdm1_ccsd_global_wf(emb, ao_basis=False, with_mf=True, t_as_lambda=Fals
+                    if spin == 'a':
+                        nonlocal total_sv_a, kept_sv_a
+                    elif spin == 'b':
+                        nonlocal total_sv_b, kept_sv_b
+                    u, s, v = np.linalg.svd(a, full_matrices=False)
+                    if svd_tol is not None:
+                        keep = (s >= svd_tol)
+                        if spin == 'a':
+                            total_sv_a += len(keep)
+                            kept_sv_a += sum(keep)
+                        elif spin == 'b':
+                            total_sv_b += len(keep)
+                            kept_sv_b += sum(keep)
+                        u, s, v = u[:,keep], s[keep], v[keep]
+                    return u, s, v
+
+                uxy_occ_a, sxy_occ_a, vxy_occ_a = svd(rxy_occ_a, 'a')
+                uxy_vir_a, sxy_vir_a, vxy_vir_a = svd(rxy_vir_a, 'a')
+                uxy_occ_a *= np.sqrt(sxy_occ_a)[np.newaxis,:]
+                uxy_vir_a *= np.sqrt(sxy_vir_a)[np.newaxis,:]
+                vxy_occ_a *= np.sqrt(sxy_occ_a)[:,np.newaxis]
+                vxy_vir_a *= np.sqrt(sxy_vir_a)[:,np.newaxis]
+
+                uxy_occ_b, sxy_occ_b, vxy_occ_b = svd(rxy_occ_b, 'b')
+                uxy_vir_b, sxy_vir_b, vxy_vir_b = svd(rxy_vir_b, 'b')
+                uxy_occ_b *= np.sqrt(sxy_occ_b)[np.newaxis,:]
+                uxy_vir_b *= np.sqrt(sxy_vir_b)[np.newaxis,:]
+                vxy_occ_b *= np.sqrt(sxy_occ_b)[:,np.newaxis]
+                vxy_vir_b *= np.sqrt(sxy_vir_b)[:,np.newaxis]
+            else:
+                nsv_a = (min(rxy_occ_a.shape[0], rxy_occ_a.shape[1])
+                     + min(rxy_vir_a.shape[0], rxy_vir_a.shape[1]))
+                total_sv_a = kept_sv_a = (total_sv_a + nsv_a)
+                nsv_b = (min(rxy_occ_b.shape[0], rxy_occ_b.shape[1])
+                     + min(rxy_vir_b.shape[0], rxy_vir_b.shape[1]))
+                total_sv_b = kept_sv_b = (total_sv_b + nsv_b)
+
+            # --- If ovlp_tol is given, the cluster x-y pairs will be screened based on the
+            # largest singular value of the occupied and virtual overlap matrices
+            total_xy += 1
+            if ovlp_tol is not None:
+                if svd_tol:
+                    rxy_occ_a_norm = (sxy_occ_a[0] if len(sxy_occ_a) > 0 else 0.0)
+                    rxy_occ_b_norm = (sxy_occ_b[0] if len(sxy_occ_b) > 0 else 0.0)
+                    rxy_vir_a_norm = (sxy_vir_a[0] if len(sxy_vir_a) > 0 else 0.0)
+                    rxy_vir_b_norm = (sxy_vir_b[0] if len(sxy_vir_b) > 0 else 0.0)
+                else:
+                    rxy_occ_a_norm = np.linalg.norm(rxy_occ_a, ord=2)
+                    rxy_occ_b_norm = np.linalg.norm(rxy_occ_b, ord=2)
+                    rxy_vir_a_norm = np.linalg.norm(rxy_vir_a, ord=2)
+                    rxy_vir_b_norm = np.linalg.norm(rxy_vir_b, ord=2)
+                if (min(rxy_occ_a_norm, rxy_occ_b_norm, rxy_vir_a_norm, rxy_vir_b_norm) < ovlp_tol):
```

Comment:
I don't think we want to take the minimum, but the maximum between spin channels.
So `min(max(occ_a, occ_b), max(vir_a, vir_b))`

Reply:
Nice catch.

Reply:
Somewhere around ~400 orbitals.

![image](https://user-images.githubusercontent.com/89935194/181313669-0472512e-44f9-4290-bdd4-88ce1e48ee23.png)